### ISSUE TITLE
[QANet-Training] Refine loss function and other training tricks

### ISSUE THREAD

**Issue by cgraywang** [dmlc/gluon-nlp#531]
-Unify loss function with basic loss
-Create exponential moving average
-Create learning rate warm up scheduler

**Comment by szha** [comment]
Consider making it part of API. Feel free to initiate API discussion here if needed.

**Comment by cgraywang** [comment]
@szha good suggestion! will discuss based an initial design.
